src/System.Net.Sockets/tests/FunctionalTests/UnixDomainSocketTest.cs

```diff
@@ -0,0 +1,221 @@
+        {
+            ManualResetEvent complete = (ManualResetEvent)args.UserToken;
+            complete.Set();
+        }
+
+        [Fact]
+        [PlatformSpecific(PlatformID.Windows)]
+        public void Socket_CreateUnixDomainSocket_Throws_OnWindows()
+        {
+            Assert.False(Capability.UnixDomainSocketsSupport());
+            Assert.Throws<SocketException>(() => new Socket(AddressFamily.Unix, SocketType.Stream, ProtocolType.Unspecified));
+        }
+
+        [Fact]
+        [PlatformSpecific(PlatformID.Linux | PlatformID.OSX)]
+        public void Socket_ConnectAsyncUnixDomainSocketEndPoint_Success()
+        {
+            Assert.True(Capability.UnixDomainSocketsSupport());
+
+            SocketTestServer server = SocketTestServer.SocketTestServerFactory(_endPoint, ProtocolType.Unspecified);
+
+            SocketAsyncEventArgs args = new SocketAsyncEventArgs();
+            args.RemoteEndPoint = _endPoint;
+            args.Completed += OnConnectAsyncCompleted;
+
+            ManualResetEvent complete = new ManualResetEvent(false);
+            args.UserToken = complete;
+
+            Socket sock = new Socket(AddressFamily.Unix, SocketType.Stream, ProtocolType.Unspecified);
+            Assert.True(sock.ConnectAsync(args));
+
+            complete.WaitOne();
+
+            Assert.Equal(SocketError.Success, args.SocketError);
+            Assert.Null(args.ConnectByNameError);
+
+            complete.Dispose();
+            sock.Dispose();
+            server.Dispose();
+        }
+
+        [Fact]
+        [PlatformSpecific(PlatformID.Linux | PlatformID.OSX)]
+        public void Socket_ConnectAsyncUnixDomainSocketEndPoint_NotServer()
+        {
+            Assert.True(Capability.UnixDomainSocketsSupport());
+
+            SocketAsyncEventArgs args = new SocketAsyncEventArgs();
+            args.RemoteEndPoint = _endPoint;
+            args.Completed += OnConnectAsyncCompleted;
+
+            ManualResetEvent complete = new ManualResetEvent(false);
+            args.UserToken = complete;
+
+            Socket sock = new Socket(AddressFamily.Unix, SocketType.Stream, ProtocolType.Unspecified);
+
+            bool willRaiseEvent = sock.ConnectAsync(args);
+            if (willRaiseEvent)
+            {
+                complete.WaitOne();
+            }
+
+            Assert.Equal(SocketError.SocketError, args.SocketError);
+
+            complete.Dispose();
+            sock.Dispose();
+        }
+
+        public void Dispose()
+        {
+            if (File.Exists(_path))
+            {
+                try
+                {
+                    File.Delete(_path);
+                }
+                catch
+                {
+                }
+            }
+
+        }
+
+        #region GC Finalizer test
+
+        // This test assumes sequential execution of tests and that it is going to be executed after other tests
+        // that used Sockets. 
```

Comment:
Good point, and I agree with you. This particular code I copied from other socket tests. They all assume sequential execution and have that finalizer test method. I don't think it's really needed for the Unix Domain Socket test, and I'd rather remove it at all. Other existing tests should cover the finalization of Socket infrastructure, and this particular fix doesn't affect any finalization at all.